nix/lib/sourcer.nix

```diff
@@ -1,13 +1,17 @@
 { lib, stdenv, git, tag ? "" }:
 let
   whitelistSource = src: allowedPrefixes:
-    builtins.filterSource
-      (path: type:
+    builtins.path {
+      filter = (path: type:
         lib.any
           (allowedPrefix:
-            lib.hasPrefix (toString (src + "/${allowedPrefix}")) path)
-          allowedPrefixes)
-      src;
+            (lib.hasPrefix (toString (src + "/${allowedPrefix}")) path) ||
+            (type == "directory" && lib.hasPrefix path (toString (src + "/${allowedPrefix}")))
```

Comment:
Just out of curiosity , why do we need type = "directory" explicitly ?

Reply:
Tbh I don't remember, I simply copied this incantation from other repos where this change was already made